packages/agent/src/hl7.ts

```diff
@@ -27,11 +27,11 @@ export class AgentHl7Channel extends BaseChannel {
       return;
     }
     this.started = true;
-    const address = new URL(this.getEndpoint().address as string);
-    const encoding = address.searchParams.get('encoding') ?? undefined;
-    const enhancedMode = address.searchParams.get('enhanced')?.toLowerCase() === 'true';
+
+    const address = new URL(this.getEndpoint().address);
     this.log.info(`Channel starting on ${address}...`);
-    this.server.start(Number.parseInt(address.port, 10), encoding, enhancedMode);
+    this.configureHl7ServerAndConnections();
```

Comment:
Centralizing all configuration into one method that we can call both on initialization and reloading of config